### DISCUSSION TITLE
Backups: user/BirdNET-Pi/BirdDB.txt?

### DISCUSSION THREAD

**Discussion by VictorioBerra** [mcguirepr89/BirdNET-Pi#244] [Q&A]
The [backups docs](https://github.com/mcguirepr89/BirdNET-Pi/wiki/Backup-and-Restore-the-Database) dont mention this file, should I back it up to?

If so, are there other files the docs do not mention, that I should back up?

**Comment by mcguirepr89** [comment]
That shouldn't be necessary. I would say that you may also want the data directory containing the extracted recordings, which is the `~/BirdSongs/Extracted/By_Date` directory

**Reply by oliverk71** [reply]
Hi @mcguirepr89, I always have to chown the By_Date directory before I can copy it to my machine. Is there another, easier way? I thought about adding a user "pi" to my PC running Linux Mint. On the other hand, I am going to make a new install anyway, so I would add the new (hard to guess) username on my PC.

**Reply by mcguirepr89** [reply]
Hi, Oliver
>I always have to chown the By_Date directory before I can copy it to my machine. Is there another, easier way?

Not that I am aware -- copying the files directly is the most straightforward way I know to recommend. Perhaps it would just be easier to put the SD card into a reader and then plug that into your PC (which will mount the files as your Mint user) and copy them that way. While the system is online, however, there is no easier way than copying the files directly.
There is, however, a much more efficient way, which it to familiarize yourself with `rsync` and setup an `rsync` daemon service to keep hardlinks synced somewhere regularly.

**Reply by oliverk71** [reply]
That's exactly they way I do it. But the files own the user pi, so I need to `chown` them. Not a big problem, though, just an extra step. 
But `rsync` seems the way to go. Thanks, Patrick.

**Reply by oliverk71** [reply]
Well, I thought about that again. Just an inspiration for future improvements: Of course, I can `rsync` by myself, but since I lost my recorded bird songs more than once in the past (because when I backupped it only saved the folders, but not the files, because of the owner problem mentioned above), I would really appreciate a built-in backup possibility where I can just provide a path or select a backup folder location. For me it would be important to not only save the birds.db but also the bird songs. Not sure, if it should be started manually or time-based, but I guess it would be nice if the user could choose. And I just thought, if the pi is connected to a network anyway, maybe the bird songs could be saved to a network location by default? 
I don't know, but I think the merge of two birds.db can be a difficult task.

**Reply by VictorioBerra** [reply]
Built in backup would be cool.

I use restic to invoke rclone which uses rsync.net as my target.

I found a GitHub repo for automating restic that worked perfectly.
